code/frameworks/svelte-vite/src/plugins/svelte-docgen.ts

```diff
@@ -73,69 +142,81 @@ export async function svelteDocgen(svelteOptions: Record<string, any> = {}): Pro
     async transform(src: string, id: string) {
       if (!filter(id)) return undefined;
 
-      if (preprocessOptions && !docPreprocessOptions) {
-        /*
-         * We can't use vitePreprocess() for the documentation
-         * because it uses esbuild which removes jsdoc.
-         *
-         * By default, only typescript is transpiled, and style tags are removed.
-         *
-         * Note: these preprocessors are only used to make the component
-         * compatible to sveltedoc-parser (no ts), not to compile
-         * the component.
-         */
-        docPreprocessOptions = [replace([[/<style.+<\/style>/gims, '']])];
+      const resource = path.relative(cwd, id);
 
-        try {
-          const ts = require.resolve('typescript');
-          if (ts) {
-            docPreprocessOptions.unshift(typescript());
+      // Get props information
+      const docgen = generateDocgen(resource, sourceFileCache);
+      const data = transformToSvelteDocParserDataItems(docgen);
+
+      let componentDoc: SvelteComponentDoc & { keywords?: string[] } = {};
+
+      if (!docgen.runePropsUsed) {
+        // Use sveltedoc-parser for generating documentation of events and slots.
```

Comment:
Can you elaborate on why this is necessary? Based on the test cases I see in `svelte2tsx`, it emits information about slots and events too?

https://github.com/sveltejs/language-tools/blob/master/packages/svelte2tsx/test/svelte2tsx/samples/component-multiple-slots/expectedv2.ts

https://github.com/sveltejs/language-tools/blob/master/packages/svelte2tsx/test/svelte2tsx/samples/component-events-interface-constant/expectedv2.ts

Reply:
`svelte-docparser` can extract slot information from HTML comments, as shown here: https://github.com/alexprey/sveltedoc-parser/issues/61. This is something `svelte2tsx` alone cannot do.

Reply:
Ah thanks for clarifying!